M2/Macaulay2/packages/FastLinAlg.m2

```diff
@@ -855,6 +940,71 @@ projDim(Module) := opts -> (N1) -> (
     return i;
 );
 
+isGBDone := (myGB) -> (
+    --a temporary function for finding out if a gb computation is done.
+    myStr := status myGB;
+    if (0 < #select("status: done", myStr)) then return true else return false;
+);
+
+
+
+isCodimAtLeast = method(Options => optIsCodimAtLeast);
+
+isCodimAtLeast(ZZ, Ideal) := opts -> (n1, I1) -> (
+    R1 := ring I1;
+    S1 := ambient R1;
+    if (not isPolynomialRing(S1)) then error "isCodimAtLeast:  This requires an ideal in a polynomial ring, or in a quotient of a polynomial ring.";
```

Comment:
Lots of redundant parentheses in these lines (and elsewhere).  Example:
```
if (codim monomialIdeal(I1) >= n1) then return true;
```
could be
```
if codim monomialIdeal I1 >= n1 then return true;
```

Reply:
I actually strongly prefer the parenethesed version.  I find it much easier to read.

Reply:
Okay, that's fine.